packages/babel-plugin-proposal-explicit-resource-management/test/fixtures/exec-async/sync-fallback-promise.js

```diff
@@ -0,0 +1,35 @@
+return (async () => {
+  const log = [];
+
+  const promiseDispose = new Promise((resolve, _) => {
+    setTimeout(() => {
+      log.push("y dispose promise body");
+      resolve();
+    }, 0);
+  });
+
+  {
+    await using x = {
+      [Symbol.asyncDispose || Symbol.for("Symbol.asyncDispose")]() {
+        log.push("x asyncDispose body");
+      },
+    };
+    await using y = {
+      [Symbol.dispose || Symbol.for("Symbol.dispose")]() {
+        log.push("y dispose body");
+        return promiseDispose;
+      },
+    };
+  }
+
+  log.push("body");
+
+  await promiseDispose;
+
+  expect(log).toEqual([
+    "y dispose body",
+    "x asyncDispose body",
+    "body",
+    "y dispose promise body",
+  ]);
```

Comment:
In current main, the `y dispose promise body` is printed before `x asyncDipose body` ([REPL](https://babel.dev/repl#?browsers=chrome%2032&build=&builtIns=false&corejs=3.21&spec=false&loose=true&code_lz=BQQwzgngdgxgBMAlHAvAPjgbwFBzjAeyjABc4AbAgc1TgG0BdAbm13yNLgAcAnAgWwCWYAKYARYVwKjaUEQHc4ABT5DRwYDxFgC5AG4iANHAD6ydFjZ5RJACqD-IggFcSG8xhx5vFagDouZzAAC2AAIgg4ABNJaRFuVWF4gCMCKIgwxBYfPC0dfREkbO8AX2MABiy2Eqq2L28QeRBBMiDBKBoAD1p6nzoAZQh-VPI_cGgYCTApGQAfWbhB4d0_ADMCHnClkbHIWCmZkUyGJEsc70oqAKDQsO7x_diZVPTM4p8yqzgS97hG5taYHaNEiKDOOQGQx2MWmcTg80WUJW602YW2Kxhh2Op16OUu1xC4UimLhLwyVXOuREJGcPCgCQESQOcV-eE-H2KJVYeHxgUJYTJb25fyaLQZanETxELDYhGIuhEfkuwEuiGwNSKQA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=&prettier=false&targets=&version=7.24.4&externalPlugins=%40babel%2Fplugin-proposal-explicit-resource-management%407.24.1&assumptions=%7B%7D))